homeassistant/components/whois/sensor.py

```diff
@@ -49,53 +49,27 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
 class WhoisSensor(SensorEntity):
     """Implementation of a WHOIS sensor."""
 
+    _attr_icon = "mdi:calendar-clock"
+    _attr_native_unit_of_measurement = TIME_DAYS
+
     def __init__(self, name, domain):
         """Initialize the sensor."""
         self.whois = whois.whois
-
-        self._name = name
         self._domain = domain
+        self._attr_name = name
 
-        self._state = None
-        self._attributes = None
-
-    @property
-    def name(self):
-        """Return the name of the sensor."""
-        return self._name
-
-    @property
-    def icon(self):
-        """Return the icon to represent this sensor."""
-        return "mdi:calendar-clock"
-
-    @property
-    def native_unit_of_measurement(self):
-        """Return the unit of measurement to present the value in."""
-        return TIME_DAYS
-
-    @property
-    def native_value(self):
-        """Return the expiration days for hostname."""
-        return self._state
-
-    @property
-    def extra_state_attributes(self):
-        """Get the more info attributes."""
-        return self._attributes
-
-    def _empty_state_and_attributes(self):
+    def _empty_value_and_attributes(self):
         """Empty the state and attributes on an error."""
-        self._state = None
-        self._attributes = None
+        self._attr_native_value = None
+        self._attr_extra_state_attributes = None
```

Comment:
Should these two not also be initialised in `__init__`?

Reply:
`update_before_add` is `True` in add_entities()